HHS/Head-Start-TTADP
# [TTAHUB-3318] Attempt to correct goalTemplateId setting bug, properly dedupe FEI, correct **some** old data

## Description of change
It seems goalTemplateId was not being propagated when goals were created via the AR. I believe I've located the source of the bug and fixed it. I've also attempted to fix some historical data that is obviously wrong, although I think Nathan is actively working on a ticket to investigate this so I'll leave the bigger cleanup to the experts.

## How to test
- Load prod data
- Visit /recipient-tta-records/7695/region/7/rttapa and confirm the goals from the support case show their root causes
- Verify there are no goals with responses and a null goalTemplateId

## Issue(s)
* https://jira.acf.gov/browse/TTAHUB-3318


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
